numpy/random/tests/test_random.py

```diff
@@ -205,12 +208,13 @@ def test_repeatability(self):
         assert_(tgt[np.dtype(np.bool).name] == res)
 
 
-class TestRandomDist(TestCase):
+class TestRandomDist:
     # Make sure the random distribution returns the correct value for a
     # given seed
 
-    def setUp(self):
-        self.seed = 1234567890
+    @classmethod
```

Comment:
At least in our code, we used generator tests heavily, then ended up swapping them back to normal tests, because they are considerably more difficult to debug.

In practice, I don't think it buys you much if one test fails but the rest in the same function still run - anyone debugging will end up fixing the first failure and then re-running the tests in any case.